### PR TITLE
[freexl] Update to 1.0.6

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  Updates freexl to 1.0.6.
  Revises the portfile and patches.
  Fixes building for uwp, removing the "skip" from CI baseline.
  Installs a pc file for windows (to be used in libspatialite etc.)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, yes (uwp builds now)

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

**Comment by dg0yt** [comment]
If someone builds this on Windows, please check the contents of the pc file.

**Comment by ras0219-msft** [comment]
Tada:
```
PS C:\src\vcpkg> gc .\installed\x86-windows\lib\pkgconfig\freexl.pc
prefix=${pcfiledir}/../..
# Package Information for pkg-config

exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include

Name: freexl
Description: a simple library extracting data from .xls (Excel BIFF) files
Version: 1.0.6
Libs: -L"${libdir}" -lfreexl
Cflags: -I"${includedir}"
Libs.private: -liconv -lcharset
```
```
PS C:\src\vcpkg> gc .\installed\x86-windows\debug\lib\pkgconfig\freexl.pc
prefix=${pcfiledir}/../..
# Package Information for pkg-config

exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/../include

Name: freexl
Description: a simple library extracting data from .xls (Excel BIFF) files
Version: 1.0.6
Libs: -L"${libdir}" -lfreexl
Cflags: -I"${includedir}"
Libs.private: -liconv -lcharset
```

File: ports/freexl/portfile.cmake [hunk 1]
```diff
@@ -1,70 +1,86 @@
...
-    set(LIBS_ALL_DBG 
-      "\"${CURRENT_INSTALLED_DIR}/debug/lib/iconv.lib\" \
-      \"${CURRENT_INSTALLED_DIR}/debug/lib/charset.lib\""
-      )
-    set(LIBS_ALL_REL 
-      "\"${CURRENT_INSTALLED_DIR}/lib/iconv.lib\" \
-      \"${CURRENT_INSTALLED_DIR}/lib/charset.lib\""
-      )
-    
+    set(OPTFLAGS "/nologo /fp:precise /W3 /D_CRT_SECURE_NO_WARNINGS /DDLL_EXPORT")
```
Comment (ras0219-msft) [inline ports/freexl/portfile.cmake:19]: I look forward to the day that we have `vcpkg_get_cmake_vars` as an official stable helper port

**Review by ras0219-msft — APPROVED** [review]
LGTM, give me a thumbs up if you're happy with the .pc files and it's ready to merge.

File: ports/freexl/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/freexl/portfile.cmake:19]: `vcpkg_build_nmake` already passes `CMAKE_CXX_FLAGS` (+debug/relase) from (directly including!) `scripts/toolchains/windows.cmake` (or chainload override) via `ENV{CL}`. It does not use `vcpkg_get_cmake_vars`.

However, it doesn't add `/LIBPATH:...`, and I wonder if it should. There are also other quirks. I considered creating a `vcpkg-nmake` port...

**Comment by dg0yt** [comment]
@ras0219-msft Thumbs up, ready to merge.

**Pull request merged**